File: src/main/java/me/sd_master92/customvoting/commands/FakeVoteCommand.java
package me.sd_master92.customvoting.commands;

import me.sd_master92.customvoting.Main;
import me.sd_master92.customvoting.VoteFile;
import me.sd_master92.customvoting.constants.Messages;
import me.sd_master92.customvoting.subjects.CustomVote;
import me.sd_master92.plugin.command.SimpleCommand;
import org.bukkit.ChatColor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class FakeVoteCommand extends SimpleCommand
{
    private final Main plugin;

    public FakeVoteCommand(Main plugin)
    {
        super(plugin, "fakevote");
        this.plugin = plugin;
    }

    @Override
    public void onCommand(CommandSender sender, String[] args)
    {
        if (args.length == 0)
        {
            if (sender instanceof Player)
            {
                fakeVote(sender.getName());
            } else
            {
                sender.sendMessage(ChatColor.RED + "- /fakevote <name>");
            }
        } else
        {
            String name = args[0];
            if (VoteFile.getByName(name, plugin) != null)
            {
                fakeVote(args[0]);
            } else
            {
                sender.sendMessage(Messages.INVALID_PLAYER.getMessage(plugin));
            }
        }
    }

    @Override
    public void onCommand(Player player, String[] args)
    {

    }

    private void fakeVote(String name)
    {
        CustomVote.create(plugin, name, "fakevote.com");
    }
}

File: src/main/kotlin/me/sd_master92/customvoting/commands/FakeVoteCommand.kt
package me.sd_master92.customvoting.commands

import me.sd_master92.core.command.SimpleCommand
import me.sd_master92.customvoting.CV
import me.sd_master92.customvoting.constants.enumerations.Message
import me.sd_master92.customvoting.constants.enumerations.PMessage
import me.sd_master92.customvoting.constants.interfaces.Voter
import me.sd_master92.customvoting.sendText
import me.sd_master92.customvoting.subjects.CustomVote
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class FakeVoteCommand(private val plugin: CV) : SimpleCommand(plugin, "fakevote")
{
    override suspend fun onCommand(sender: CommandSender, args: Array<out String>)
    {
        if (args.isEmpty())
        {
            if (sender is Player)
            {
                fakeVote(sender.getName(), PMessage.FAKE_VOTE_VALUE_WEBSITE.toString())
            } else
            {
                sender.sendMessage(PMessage.FAKE_VOTE_MESSAGE_COMMAND_USAGE.toString())
            }
        } else
        {
            val name = args[0]
            val service = if (args.size >= 2) args[1] else PMessage.FAKE_VOTE_VALUE_WEBSITE.toString()
            if (Voter.getByName(plugin, name) != null)
            {
                fakeVote(name, service)
            } else
            {
                sender.sendText(plugin, Message.INVALID_PLAYER)
            }
        }
    }

    override suspend fun onCommand(player: Player, args: Array<out String>)
    {
    }

    private fun fakeVote(name: String, service: String)
    {
        CustomVote.create(plugin, name, service)
    }

    init
    {
        withNoPermMessage(Message.NO_PERMISSION.getMessage(plugin))
    }
}